src/components/form/form.tsx

```diff
@@ -1,12 +1,24 @@
-import React, { FunctionComponent, ReactNode, HTMLAttributes } from 'react';
+import React, {
+  FunctionComponent,
+  ReactNode,
+  HTMLAttributes,
+  FormHTMLAttributes,
+} from 'react';
 import classNames from 'classnames';
 import { EuiCallOut } from '../call_out';
 import { EuiI18n } from '../i18n';
-import { CommonProps } from '../common';
+import { CommonProps, ExclusiveUnion } from '../common';
 
 export type EuiFormProps = CommonProps &
-  HTMLAttributes<HTMLDivElement> & {
+  ExclusiveUnion<
+    { component: 'form' } & FormHTMLAttributes<HTMLFormElement>,
+    { component?: 'div' } & HTMLAttributes<HTMLDivElement>
+  > & {
     isInvalid?: boolean;
+    /**
+     * Which HTML element to render
```

Comment:
Just because our props docs table gets confused about the possible values of props used in exclusive unions like this, let's make it easy on consumers to find the acceptable values.

```suggestion
     * Which HTML element to render `div` or `form`
```